homeassistant/components/sensor/forecast.py

```diff
@@ -124,80 +116,66 @@ def state(self):
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
-        return self._unit_of_measurement
+        self.forecast_data.update_unit_of_measurement(self.type)
```

Comment:
So once I looked at this, I realized that unit of measurement should NOT be on the data object as it is sensor type specific. So I moved it back up to the sensor level, but still implemented your recommendation and cleaned up some other things.

I think it makes more sense now. Let me know what you think @kellerza when I push in a minute.